Mathlib/Analysis/NormedSpace/Normalized.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2020 Ilmārs Cīrulis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ilmārs Cīrulis, Alex Meiburg
+-/
+import Mathlib.Analysis.RCLike.Basic
+
+/-!
+# Normalized vector
+
+Function that calculates unit length vector from a vector
+(if the given vector is nonzero vector) or returns zero vector
+(if the given vector is zero vector).
+-/
+
+variable {V : Type*}
+variable [NormedAddCommGroup V]
+variable [NormedSpace ℝ V]
+
+/-- The normalized vector from a given vector. `normalized 0 = 0`, otherwise it is
+the corresponding unit length vector. -/
+noncomputable def normalized (x : V) : V := ‖x‖⁻¹ • x
+
+@[simp]
+theorem normalized_zero_eq_zero : normalized (0 : V) = 0 := by
+  simp [normalized]
+
```

Comment:
This is a handy lemma, especially because it requires no hypotheses:
```suggestion
@[simp]
theorem normalized_zero_eq_zero : normalized (0 : V) = 0 := by
  simp [normalized]

@[simp]
theorem norm_smul_normalized (x : V) : ‖x‖ • normalized x = x := by
  by_cases hx : x = 0
  all_goals simp [normalized, hx]

```